src/main/java/Duke.java

```diff
@@ -1,10 +1,144 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.lang.Math;
+
 public class Duke {
-    public static void main(String[] args) {
+    public static ArrayList<Task> taskList = new ArrayList<>();
+    static int tasksDone = 0;
+    public static void main(String[] args) throws DukeException {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        String helpStr = "What can I help you with?";
+        String byeStr = "Bye. Hope to see you again soon!";
+
+        System.out.println(helpStr);
+        // create a Scanner object to read user input
+        Scanner input = new Scanner(System.in);
+
+        // loop until "bye"
+        while (true) {
+            // read user input
+            String command = input.nextLine();
+
+            // check if user inputs "bye"
+            if (command.equalsIgnoreCase("bye")) {
+                System.out.println(byeStr);
+                return;
+            }
+            // check if user inputs "list" and display items in list
+            else if (command.equalsIgnoreCase("list")) {
+                if (taskList.size() == 0) {
+                    System.out.println("Your list is empty!");
+                } else {
+                    for (int i = 0; i < taskList.size(); i++) {
+                        System.out.println(i + 1 + ". " + taskList.get(i));
+                    }
+                }
+            } else if (command.startsWith("mark ") || command.startsWith("unmark ")) {
+                try {
+                    // Mark task as done
+                    if (command.substring(0, command.indexOf(" ")).equalsIgnoreCase("mark")) {
+                        int index = Integer.parseInt(command.replaceAll("mark ", "")) - 1;
+                        if (taskList.get(index).isDone()) {
+                            throw new DukeException("marked");
+                        }
+                        else {
```

Comment:
should the line ```` else { ```` be placed on the same line with the preceding `}`, according to the code standard for if-else statement? :frowning: